src/common_functions.cpp

```diff
@@ -0,0 +1,291 @@
+int32_t to32(uint32_t c) { return (int32_t)c; }
+
+uint32_t to32u(char *c) { return balign32u__internal(c); }
+uint32_t to32u(unsigned char *c) { return balign32u__internal((char *)c); }
+uint32_t to32u(int32_t c) { return *(uint32_t *)c; }
+uint32_t to32u(uint32_t c) { return c; }
+
+// 16-bit wide functions
+
+int16_t bswap16__internal(char *c) {
+  int16_t *d16;
+  char d[2];
+  d[0] = c[1];
+  d[1] = c[0];
+  d16 = (int16_t *)d;
+  return *d16;
+}
+
+uint16_t bswap16u__internal(char *c) {
+  uint16_t *d16;
+  char d[2];
+  d[0] = c[1];
+  d[1] = c[0];
+  d16 = (uint16_t *)d;
+  return *d16;
+}
+
+int16_t balign16__internal(char *c) {
+  int16_t *d16;
+  char d[2];
+  d[0] = c[0];
+  d[1] = c[1];
+  d16 = (int16_t *)d;
+  return *d16;
+}
+
+uint16_t balign16u__internal(char *c) {
+  uint16_t *d16;
+  char d[2];
+  d[0] = c[0];
+  d[1] = c[1];
+  d16 = (uint16_t *)d;
+  return *d16;
+}
+
+int16_t qpx_bswap16(char *c) { return bswap16__internal(c); }
+int16_t qpx_bswap16(unsigned char *c) { return bswap16__internal((char *)c); }
+int16_t qpx_bswap16(int16_t c) { return bswap16__internal((char *)&c); }
+int16_t qpx_bswap16(uint16_t c) { return bswap16__internal((char *)&c); }
+uint16_t qpx_bswap16u(char *c) { return bswap16u__internal(c); }
+uint16_t qpx_bswap16u(unsigned char *c) {
+  return bswap16u__internal((char *)c);
+}
+uint16_t qpx_bswap16u(int16_t c) { return bswap16__internal((char *)&c); }
+uint16_t qpx_bswap16u(uint16_t c) { return bswap16__internal((char *)&c); }
+
+int16_t to16(char *c) { return balign16__internal(c); }
+int16_t to16(unsigned char *c) { return balign16__internal((char *)c); }
+int16_t to16(int16_t c) { return c; }
+int16_t to16(uint16_t c) { return (int16_t)c; }
+uint16_t to16u(char *c) { return balign16u__internal(c); }
+uint16_t to16u(unsigned char *c) { return balign16u__internal((char *)c); }
+uint16_t to16u(int16_t c) { return (uint16_t)c; }
+uint16_t to16u(uint16_t c) { return c; }
+
+void lba2msf(int lba, msf *time) {
+  time->m = lba / 4500;
+  time->s = (lba / 75) % 60;
+  time->f = lba % 75;
+}
+
+int msf2lba(msf time) { return time.m * 4500 + time.s * 75 + time.f; }
+
+void int2hms(int intt, hms *time) {
+  time->h = intt / 3600;
+  time->m = (intt / 60) % 60;
+  time->s = intt % 60;
+}
+
+#ifndef _WIN32
+
+int min(int a, int b) {
+  if (a <= b)
+    return a;
+  return b;
+}
+
+int max(int a, int b) {
+  if (a > b)
+    return a;
+  return b;
+}
+
+#else
+
+int inet_aton(const char *cp, struct in_addr *addr) {
+  addr->s_addr = inet_addr(cp);
+  return 1;
+}
+
+#endif
+
+void remove_double_spaces(char *str) {
+  int len = strlen(str);
+  while (len--) {
+    if (*str == 0x20 && str[1] == 0x20)
+      strcpy(str, str + 1);
+    else
+      str++;
+  }
+}
+
+void remove_end_spaces(char *str) {
+  int len = strlen(str);
+  while (str[--len] == 0x20)
+    str[len] = 0;
+}
+
+int dispers(int m, int *arr, int l) {
+  if ((!m) || (!arr) || (l < 2))
+    return 0;
+  long disp = 0;
+  int i;
+  int q;
```

Comment:
Codacy found an issue: [The scope of the variable 'q' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)